rules/options.go

```diff
@@ -74,6 +76,8 @@ func makeEngineOptions(options ...EngineOption) engineOptions {
 		syncInterval:           300,
 		syncGetTimeout:         0,
 		watchTimeout:           0,
+		keyProcConcurrency:     5,
```

Comment:
Any metrics needed around this new channel to gain more insight on key protect operations?